relays/client-substrate/src/finality_source.rs

```diff
@@ -31,22 +32,27 @@ use std::{marker::PhantomData, pin::Pin};
 
 /// Wrapped raw Justification.
 #[derive(Debug, Clone)]
-pub struct Justification<Number> {
+pub struct Justification<H: HeaderT> {
 	/// Header number decoded from the [`raw_justification`].
-	target_header_number: Number,
-	/// Raw, encoded justification bytes.
-	raw_justification: sp_runtime::Justification,
+	target_header_number: H::Number,
+	/// Decoded GRANDPA justification.
+	justification: GrandpaJustification<H>,
 }
 
-impl<Number> Justification<Number> {
-	/// Extract raw justification.
-	pub fn into_inner(self) -> sp_runtime::Justification {
-		self.raw_justification
+impl<H: HeaderT> Justification<H> {
+	/// Extract raw encoded justification.
```

Comment:
This method is not used anywhere && you can get the same result with `.justification().encode()`, so there's no reason to keep it imo.